src/metadata/metadataVisitor.ts

```diff
@@ -17,21 +35,27 @@ export function metadataVisitor(
       if (!decorators || decorators.length === 0) return;
 
       decorators!.push(
-        t.decorator(
-          t.callExpression(
-            t.memberExpression(
-              t.identifier('Reflect'),
-              t.identifier('metadata')
-            ),
-            [
-              t.stringLiteral('design:paramtypes'),
-              t.arrayExpression(
-                field.params.map(param => serializeType(classPath, param))
-              )
-            ]
+        createMetadataDesignDecorator(
+          'design:type',
+          t.identifier('Function')
+        )
+      );
+      decorators!.push(
+        createMetadataDesignDecorator(
+          'design:paramtypes',
+          t.arrayExpression(
+            field.params.map(param => serializeType(classPath, param))
           )
         )
       );
+      // decorators!.push(
```

Comment:
Please, add a comment here warning that following code is just an hint for future works about `returnType`: elseway this seems just like a forgotten copy paste 😄

Reply:
I added a shorter comment with a useful reference to the TSC compiler. If someone wants to tackle this at some time, this will be a good starting point.